apps/jobs/views.py

```diff
@@ -355,20 +365,20 @@ def leaderboard(request, challenge_phase_split_id):
 
     leaderboard_data = leaderboard_data.annotate(
         filtering_score=RawSQL('result->>%s', (default_order_by, ), output_field=FloatField())).values(
-            'id', 'submission__participant_team__team_name',
+            'id', 'submission__participant_team__team_name', 'submission__baseline_submission',
             'challenge_phase_split', 'result', 'filtering_score', 'leaderboard__schema', 'submission__submitted_at')
 
     sorted_leaderboard_data = sorted(leaderboard_data, key=lambda k: float(k['filtering_score']), reverse=True)
-
+    # challenge_host = challenge_phase_split__
     distinct_sorted_leaderboard_data = []
-    team_list = []
-
+    team_list = set()
     for data in sorted_leaderboard_data:
         if data['submission__participant_team__team_name'] in team_list:
             continue
         else:
```

Comment:
I think adding this condition and removing the condition that you have added will do the thing.
```
        elif data['submission__baseline_submission'] == True:
            distinct_sorted_leaderboard_data.append(data)
        else:
            distinct_sorted_leaderboard_data.append(data)
            team_list.append(data['submission__participant_team__team_name'])
```